pam/pam.go

```diff
@@ -69,7 +82,23 @@ func sendReturnMessageToPam(mTx pam.ModuleTransaction, retStatus adapter.PamRetu
 
 // Authenticate is the method that is invoked during pam_authenticate request.
 func (h *pamModule) Authenticate(mTx pam.ModuleTransaction, flags pam.Flags, args []string) error {
-	return h.handleAuthRequest(authd.SessionMode_AUTH, mTx, flags, args)
+	// Do not try to start authentication again if we've been already through this.
+	_, err := mTx.GetData(authenticatedKey)
+	if err == nil && parseArgs(args)["force_reauth"] != "true" {
+		return pam.ErrIgnore
+	}
+	if err != nil && !errors.Is(err, pam.ErrNoModuleData) {
+		return err
+	}
+
+	err = h.handleAuthRequest(authd.SessionMode_AUTH, mTx, flags, args)
+
```

Comment:
No need for this whitespace. The check below evaluates the returned error from h.handleAuthRequest, so it should be scoped with the function call.